File: Programs/LC 13/LC_13_1.py
import os
cwd = os.path.dirname(__file__)

class Solution:
    def romanToInt(self, s: str) -> int:
        romanNumerals = {
            'M': 1000, 
            'D': 500, 
            'C': 100, 
            'L': 50, 
            'X': 10, 
            'V': 5, 
            'I': 1
        }

        sum, prevNum = 0, -1

        for romanNumeral in s:
            rn = romanNumerals[romanNumeral]

            if prevNum == -1:
                sum = rn
            else:
                if prevNum < rn:
                    sum -= prevNum
                    sum += (rn - prevNum)
                else:
                    sum += rn

            prevNum = rn
        
        return sum

if __name__ == '__main__': 
    nInputs = 6
    solObj = Solution()

    for i in range(1, nInputs + 1): 
        fname = "input{number}.txt"
        fpath = os.path.join(cwd, fname.format(number = i))
        
        print("======== TestCase {number} ========".format(number = i))
        try: 
            with open(fpath, "r") as fileHandle: 
                #  ===== START =====
                s = fileHandle.readline()

                print(solObj.romanToInt(s))

                #  ===== END =====
        except FileNotFoundError:
            print("Cannot Open the test case")
            break

File: Programs/LC 13/LC_13_1.cpp
#include<iostream>
#include<fstream>
#include<vector>
#include<sstream>

using namespace std;

int romanToInt(string s) {
    int sum = 0; 
    int prevNum = -1; 

    for(int i = 0; i < s.length(); ++i){
        int rn = 0; 
        switch(s[i]){
            case 'M': rn = 1000; break; 
            case 'D': rn = 500; break; 
            case 'C': rn = 100; break; 
            case 'L': rn = 50; break; 
            case 'X': rn = 10; break; 
            case 'V': rn = 5; break; 
            case 'I': rn = 1; break;
        }

        if(prevNum == -1){
            sum = rn; 
        }else{
            if(prevNum < rn){
                sum -= prevNum; 
                sum += (rn - prevNum);
            }else
                sum += rn; 
        }
        prevNum = rn; 
    }

    return sum; 
}

int main(){
    // change the number of test cases
    int nInputs = 6;

    // test case files should be input1.txt, input2.txt, ..., inputN.txt format
    for(int i = 1; i <= nInputs; i++){
        string filename = "input" + to_string(i) + ".txt"; 

        ifstream fin(filename); 
        if(!fin.is_open()){
            cout<<"Cannot Open the test case"<<endl;
            return 0; 
        }

        cout<<"======== TestCase "<<i<<" ========"<<endl;


        // Write input and output here
        // Instead of cin use fin

        // ===== START =====
        string s; fin>>s; 

        cout<<romanToInt(s)<<endl; 
        
        // ===== END =====
    }
    
    return 0; 
}